_ml-commons-plugin/remote-models/async-batch-ingestion.md

```diff
@@ -0,0 +1,190 @@
+
+```
+{"SageMakerOutput":[[-0.017166402,0.055771016,...],[-0.06422759,-0.004301484,...],"content":["this is chapter 1","harry potter"],"id":1}
+{"SageMakerOutput":[[-0.017455402,0.023771016,...],[-0.02322759,-0.009101284,...],"content":["this is chapter 2","draco malfoy"],"id":1}
+...
+```
+
+## Ingesting data from a single file
+
+First, create a k-NN index where you'll ingest the data. The fields in the k-NN index represent the structure of the data in the source file. 
+
+In this example, the source file contains documents containing titles and chapters, along with their corresponding embeddings. Thus, you'll create a k-NN index with fields `id`, `chapter_embedding`, `chapter`, `title_embedding`, and `title`:
+
+```json
+PUT /my-nlp-index
+{
+  "settings": {
+    "index.knn": true
+  },
+  "mappings": {
+    "properties": {
+      "id": {
+        "type": "text"
+      },
+      "chapter_embedding": {
+        "type": "knn_vector",
+        "dimension": 384,
+        "method": {
+          "engine": "nmslib",
+          "space_type": "cosinesimil",
+          "name": "hnsw",
+          "parameters": {
+            "ef_construction": 512,
+            "m": 16
+          }
+        }
+      },
+      "chapter": {
+        "type": "text"
+      },
+      "title_embedding": {
+        "type": "knn_vector",
+        "dimension": 384,
+        "method": {
+          "engine": "nmslib",
+          "space_type": "cosinesimil",
+          "name": "hnsw",
+          "parameters": {
+            "ef_construction": 512,
+            "m": 16
+          }
+        }
+      },
+      "title": {
+        "type": "text"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+When using an S3 file as the source for asynchronous batch ingestion, you must map the fields in the source file to fields in the index in order to indicate where each piece of data is ingested. If no JSON path is provided for a field, that field in the k-NN index will be set to `null`.
+
+In the `field_map`, provide the location where the data for each field can be found in the source file. You can also specify fields to be ingested directly into your index without making any changes to the source file by adding their JSON path to the `ingest_fields` array. For example, in the following asynchronous batch ingestion request, the element with the JSON path `$.id` from the source file is ingested directly into the `id` field of your index. To ingest this data from the Amazon S3 file, send the following request to your OpenSearch endpoint:
+
```

Comment:
Above, third sentence: "the source file element with the JSON path `$.id` is ingested..."?